pkg/plugins/autodiscovery/dockerfile/dockerfile.go

```diff
@@ -19,8 +19,178 @@ var (
 		"Dockerfile",
 		"Dockerfile.*",
 	}
+	// GlobalIgnore specifies a list of globally ignored docker image, like the `scratch` image which
+	// never needs to be updated as it's not a "real" image
+	ScratchIgnore MatchingRule = MatchingRule{
+		Images: []string{"scratch"},
+	}
 )
 
+func (d Dockerfile) generateManifest(
+	globalIgnore MatchingRules,
+	relativeFoundDockerfile, basename, dirname string,
+	instruction keywords.FromToken,
+	args map[string]keywords.SimpleTokens,
+) ([]byte, error) {
+	var err error
+	targetMatcher, targetInstruction := instruction.Image, instruction.Keyword
+	image, tag, digest, platform := instruction.Image, instruction.Tag, instruction.Digest, instruction.Platform
+	for arg_type, fromArg := range instruction.Args {
+		if arg, ok := args[fromArg.Name]; ok {
+			switch arg_type {
+			case "image":
+				image = strings.Replace(image, "${"+fromArg.Name+"}", arg.Value, -1)
+			case "tag":
+				if arg.Value == "" {
+					// Not enough information, should quit
+					return nil, nil
```

Comment:
Wouldn't it be better to have a "continue" instead of a "return" so we also process the other args defined by `instruction.Args`

Reply:
Agreed, I've still handle the case where an arg is not defined and doesn't allow us to define a manifest, as stated here https://github.com/updatecli/updatecli/blob/main/pkg/plugins/autodiscovery/dockerfile/testdata/updatecli-action/Dockerfile#L2-L3